industrial_ci/util.sh

```diff
@@ -51,93 +51,76 @@ function ici_time_start {
-        ici_time_end $color_wrap
-    else
-	echo "Previous Travis fold name not found. It might be either successful termination of the script, or wrong call. Skipping 'ici_time_end' anyway."
+    # end fold if needed
+    if [ -n "$TRAVIS_FOLD_NAME" ]; then
+        if [ $exit_code -ne "0" ]; then color_wrap=31; fi  # Red color for errors
+        ici_time_end "$color_wrap" "$exit_code"
     fi
 
-    if [ $exit_code -eq "1" ]; then trap - ERR; fi  # Reset signal handler since the shell is about to exit.
-    if [ "$DEBUG_BASH" ] && [ "$DEBUG_BASH" == true ]; then set -x; fi
-    if [ $exit_code -ne "-1" ]; then exit $exit_code; fi
+    if [ "$exit_code" == "${EXPECT_EXIT_CODE:-0}" ]; then
+        exit 0
+    elif [ "$exit_code" == "0" ]; then # 0 was not expected
+        exit 1
+    fi
+
+    exit $exit_code
 }
 
 #######################################
-# This calls "exit 1", along with the following. When your script on Travis CI already uses other functions from this file (util.sh), using this is recommended over calling directly "exit 1".
+# Print an error message and calls "exit"
 #
 # * wraps the section that is started by ici_time_start function with the echo color red (31).
 # * reset signal handler for ERR to the bash default one. Subsequent signal handlers for ERR if any are unaffected by any handlers defined prior.
 #
 # Globals:
 #   (None)
 # Arguments:
-#   (None)
+#   message (optional) 
+#   exit_code (default: $?)
 # Returns:
 #   (None)
 #######################################
 function error {
-    _end_fold_script 1 31
-}
-
-#######################################
-# Similar to `error` function, this lets you "exit 0" and take care of other things as following, when your script on Travis CI already uses other functions from this file (util.sh).
-#
-# * wraps the section that is started by ici_time_start function with the echo color green.
-# * reset signal handler for ERR to the bash default one. Subsequent signal handlers for ERR if any are unaffected by any handlers defined prior.
-#
-# Globals:
-#   (None)
-# Arguments:
-#   _exit_code (default: 0): Unix signal. If -1 passed then the process continues without exiting.
-# Returns:
-#   (None)
-#######################################
-function success {
-    _FUNC_MSG_PREFIX="[fuction success]"
-    _exit_code=${1:-0}  # If 1st arg is not passed, set 0.
-    HIT_ENDOFSCRIPT=${HIT_ENDOFSCRIPT:-false}
-    if [ $HIT_ENDOFSCRIPT = false ]; then
-    	if [ $_exit_code -eq 0 ]; then
-    	    echo "${_FUNC_MSG_PREFIX} Arg HIT_ENDOFSCRIPT must be true when this function exit with 0. Turn _exit_code to 1."; _exit_code=1;
-    	else
-    	    echo "${_FUNC_MSG_PREFIX} _exit_code cannot be 0 for this func. Make sure you are calling this in a right context."; _exit_code=1;
-    	fi
+    local exit_code=${2:-$?} #
+    if [ -n $1 ]; then
+        echo "\e[31m$1\e0m" # print error in red
+    fi
+    if [ "$exit_code" == "0" ]; then # 0 is not error
```

Comment:
exit_code is read from $?, which might be 0 in some cases, like `if ! false ; then echo "$?"; fi`.
So the following code works now as expected:
```bash
if ! [ -e "$my_path" ]; then
  error "path does not exist: '$my_path'"
fi
```
I think, the documentation needs to be updated..